File: app/src/main/java/com/amuyu/melitos/Adapters/VentasAdapter.java
package com.amuyu.melitos.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.amuyu.melitos.R;
import com.amuyu.melitos.ui.vender.VentasItem;

import java.util.List;

import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import static com.facebook.FacebookSdk.getApplicationContext;

public class VentasAdapter extends RecyclerView.Adapter<VentasAdapter.ItemViewHolder>{
    public static final int SPAN_COUNT_ONE = 1;
    public static final int SPAN_COUNT_THREE = 3;

    private static final int VIEW_TYPE_SMALL = 1;
    private static final int VIEW_TYPE_BIG = 2;

    private List<VentasItem> mItems;
    private GridLayoutManager mLayoutManager;

    public VentasAdapter(List<VentasItem> items, GridLayoutManager layoutManager) {
        mItems = items;
        mLayoutManager = layoutManager;
    }

    @Override
    public int getItemViewType(int position) {
        int spanCount = mLayoutManager.getSpanCount();
        if (spanCount == SPAN_COUNT_ONE) {
            return VIEW_TYPE_BIG;
        } else {
            return VIEW_TYPE_SMALL;
        }
    }

    @Override
    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view;
        if (viewType == VIEW_TYPE_BIG) {
            view = LayoutInflater.from(parent.getContext()).inflate(R.layout.vender_item, parent, false);
        } else {
            view = LayoutInflater.from(parent.getContext()).inflate(R.layout.vender_item2, parent, false);
        }
        return new ItemViewHolder(view, viewType);
    }

    @Override
    public void onBindViewHolder(ItemViewHolder holder, int position) {
        //VentasItem item = mItems.get(position);
//        String a = item.getTitle();
//        holder.title.setText("" + a);
//        holder.iv.setImageResource(item.getImgResId());
//        holder.info.setText(item.getPrice());
//        if (getItemViewType(position) == VIEW_TYPE_BIG) {
//            holder.info.setText(item.getPrice());
//        }

        //CODIGO PARA VERIFICAR LOS CHECKBOX
//        holder.checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
//                Toast toast = Toast.makeText(compoundButton.getContext(), position, Toast.LENGTH_SHORT);
//                toast.setMargin(50,50);
//                toast.show();
//            }
//        });

    }

    @Override
    public int getItemCount() {
        return 30;
    }

    class ItemViewHolder extends RecyclerView.ViewHolder{
        ImageView iv;
        TextView title;
        TextView info;
        //CheckBox checkBox;

        ItemViewHolder(View itemView, int viewType) {
            super(itemView);
            if (viewType == VIEW_TYPE_BIG) {
                iv = (ImageView) itemView.findViewById(R.id.rowImageView);
                title = (TextView) itemView.findViewById(R.id.rowTextView);
                info = (TextView) itemView.findViewById(R.id.rowTextView2);
                //checkBox = (CheckBox) itemView.findViewById(R.id.rowCheckBox);
            } else {
                iv = (ImageView) itemView.findViewById(R.id.SrowImageView);
                title = (TextView) itemView.findViewById(R.id.SrowTextView);
                info = (TextView) itemView.findViewById(R.id.SrowTextView2);
            }
        }
    }



}

File: app/src/main/java/com/amuyu/melitos/Adapters/VentasAdapter.kt
package com.amuyu.melitos.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import com.amuyu.melitos.R
import com.amuyu.melitos.ui.vender.VentasItem
import com.bumptech.glide.Glide

import java.util.ArrayList

import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.amuyu.melitos.C

import com.facebook.FacebookSdk.getApplicationContext
import kotlinx.android.synthetic.main.vender_item.view.*
import kotlinx.android.synthetic.main.vender_item2.view.*

class VentasAdapter(private val mLayoutManager: GridLayoutManager) :
    RecyclerView.Adapter<VentasAdapter.ItemViewHolder>() {
    var sw = false
    var items = arrayListOf<ItemViewHolder>()
    private var mItems: ArrayList<Map<String, Any>>? = null
    fun setmItems(data:ArrayList<Map<String, Any>>) {
        mItems = data
    }
    fun at(i: Int){

    }

    override fun getItemViewType(position: Int): Int {
        val spanCount = mLayoutManager.spanCount
        return if (spanCount == SPAN_COUNT_ONE) {
            VIEW_TYPE_BIG
        } else {
            VIEW_TYPE_SMALL
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val view: View
        if (viewType == VIEW_TYPE_BIG) {
            view = LayoutInflater.from(parent.context).inflate(R.layout.vender_item, parent, false)
        } else {
            view = LayoutInflater.from(parent.context).inflate(R.layout.vender_item2, parent, false)
        }
        return ItemViewHolder(view, viewType)
    }

    override fun onBindViewHolder(v: ItemViewHolder, position: Int) {
        v.title.text = mItems!![position].get(C.cPName).toString()
        v.info.text = mItems!![position].get(C.cPStock).toString()
        if(v.price != null){
            v.price.text = mItems!![position].get(C.cPPrice).toString()
        }
        if(sw){
            v.cv.visibility = View.VISIBLE
        }else{
            v.cv.visibility = View.GONE
        }
        items.add(position, v)
    }

    override fun getItemCount(): Int {
        if(mItems == null){return 0}
        return mItems!!.size
    }

    inner class ItemViewHolder


        (itemView: View, viewType: Int) : RecyclerView.ViewHolder(itemView) {
        var iv: ImageView
        var title: TextView
        var info: TextView
        lateinit var price: TextView
        var cv: CheckBox
        init {
            if (viewType == VIEW_TYPE_BIG) {
                iv = itemView.findViewById<View>(R.id.rowImageView) as ImageView
                title = itemView.findViewById<View>(R.id.rowTextView) as TextView
                info = itemView.findViewById<View>(R.id.rowTextView2) as TextView
                price = itemView.rowTextView3
                cv= itemView.rowCheckBox
                //checkBox = (CheckBox) itemView.findViewById(R.id.rowCheckBox);
            } else {
                iv = itemView.SrowImageView
                title = itemView.SrowTextView
                info = itemView.SrowTextView2
                cv=itemView.checkBox
            }
        }
    }

    companion object {
        val SPAN_COUNT_ONE = 1
        val SPAN_COUNT_THREE = 3

        private val VIEW_TYPE_SMALL = 1
        private val VIEW_TYPE_BIG = 2
    }


}
